server/chess.py

```diff
@@ -209,8 +199,7 @@ def get_knight_moves(
                 for file_direction in (-1, 1):
                     rank = knight.rank + rank_absolute * rank_direction
                     file = knight.file + file_absolute * file_direction
-                    victim = self.get_piece(rank, file)
-                    if (not victim) or (victim.side != knight.side):
+                    if validate_move_piece(self, knight, rank, file):
```

Comment:
```suggestion
                    if self.validate_move_piece(knight, rank, file):
```
If you must use this method, but it stills seems needlessly inefficient.